Loading .emd: AssertionError: Dataset location is ambiguous!

Hi, I am trying to load STEM videos recorded with Velox (last updated in January, I think) as .emd files. All my files gives the same error, here's a [sample file](https://polybox.ethz.ch/index.php/s/QwCBfdgZaTRjyyf).

`s = hs.load('STEM HAADF 1613.emd')` gives: 

``` 
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-14-bbbc643c53d6> in <module>()
----> 1 s = hs.load('STEM HAADF 1613.emd')

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\hyperspy\io.py in load(filenames, signal_type, stack, stack_axis, new_axis_name, lazy, **kwds)
    241             objects = [load_single_file(filename, lazy=lazy,
    242                                         **kwds)
--> 243                        for filename in filenames]
    244 
    245         if len(objects) == 1:

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\hyperspy\io.py in <listcomp>(.0)
    241             objects = [load_single_file(filename, lazy=lazy,
    242                                         **kwds)
--> 243                        for filename in filenames]
    244 
    245         if len(objects) == 1:

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\hyperspy\io.py in load_single_file(filename, signal_type, **kwds)
    284                                 reader=reader,
    285                                 signal_type=signal_type,
--> 286                                 **kwds)
    287 
    288 

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\hyperspy\io.py in load_with_reader(filename, reader, signal_type, **kwds)
    293     lazy = kwds.get('lazy', False)
    294     file_data_list = reader.file_reader(filename,
--> 295                                         **kwds)
    296     objects = []
    297 

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\hyperspy\io_plugins\emd.py in file_reader(filename, log_info, lazy, **kwds)
    407 def file_reader(filename, log_info=False,
    408                 lazy=False, **kwds):
--> 409     emd = EMD.load_from_emd(filename, lazy)
    410     if log_info:
    411         emd.log_info()

~\AppData\Local\Continuum\Anaconda3\lib\site-packages\hyperspy\io_plugins\emd.py in load_from_emd(cls, filename, lazy)
    321         # One node should remain, the data node (named 'data', 'signals',
    322         # 'experiments', ...)!
--> 323         assert len(node_list) == 1, 'Dataset location is ambiguous!'
    324         data_group = emd_file.get(node_list[0])
    325         if data_group is not None:

AssertionError: Dataset location is ambiguous!

```

Help is much appreciated :)

--- a/hyperspy/io_plugins/emd.py
+++ b/hyperspy/io_plugins/emd.py
@@ -480,7 +480,7 @@ def _get_keys_from_group(group):
 
 
 def _parse_sub_data_group_metadata(sub_data_group):
-    metadata_array = sub_data_group['Metadata'][:].T[0]
+    metadata_array = sub_data_group['Metadata'][:, 0].T
     mdata_string = metadata_array.tostring().decode("utf-8")
     return json.loads(mdata_string.rstrip('\x00'))
 
@@ -552,7 +552,7 @@ class FeiEMDReader(object):
         elif select_type is None:
             pass
         else:
-            raise ValueError("`select_type` parameter takes only: `None`, ",
+            raise ValueError("`select_type` parameter takes only: `None`, "
                              "'single_spectrum', 'images' or 'spectrum_image'.")
 
         if self.im_type == 'Image':
@@ -681,7 +681,11 @@ class FeiEMDReader(object):
                     chunks=h5data.chunks),
                 axes=[2, 0, 1])
         else:
-            data = np.rollaxis(np.array(h5data), axis=2)
+            # Preallocate the numpy array and use read_direct method, which is
+            # much faster in case of chunked data.
+            data = np.empty(h5data.shape)
+            h5data.read_direct(data)
+            data = np.rollaxis(data, axis=2)
 
         pix_scale = original_metadata['BinaryResult'].get(
             'PixelSize', {'height': 1.0, 'width': 1.0})
@@ -791,6 +795,8 @@ class FeiEMDReader(object):
         self.original_metadata.update({group_name: d})
 
     def _read_spectrum_stream(self):
+        if not self.load_SI:
+            return
         self.detector_name = 'EDS'
         # Try to read the number of frames from Data/SpectrumImage
         try:

--- a/hyperspy/io_plugins/emd.py
+++ b/hyperspy/io_plugins/emd.py
@@ -681,6 +681,8 @@ class FeiEMDReader(object):
                     chunks=h5data.chunks),
                 axes=[2, 0, 1])
         else:
+            # Workaround for a h5py bug https://github.com/h5py/h5py/issues/977
+            # Change back to standard API once issue #977 is fixed.
             # Preallocate the numpy array and use read_direct method, which is
             # much faster in case of chunked data.
             data = np.empty(h5data.shape)

